src/clojure_lsp/crawler.clj

```diff
@@ -167,12 +168,56 @@
     (concat (diagnose-unused-aliases uri declared-aliases unused-aliases)
             (diagnose-unused-references uri declared-references all-envs))))
 
-(defn find-diagnostics [project-aliases uri usages]
-  (let [unknown (diagnose-unknown project-aliases usages)
+(defn- kondo-finding->diagnostic [lines {:keys [message level row col] :as finding}]
+  (let [decrow (max 0 (dec row))
+        deccol (max 0 (dec col))
+        ^String line (when (> (count lines) decrow)
+                       (nth lines decrow))
+        start-char (when (> (count line) deccol)
+                     (.charAt line deccol))
+        expression? (identical? \( start-char)
+        finding (merge {:end-row row :end-col col} finding)]
+    {:range (shared/->range (if expression?
+                              (assoc finding :end-row row :end-col col)
+                              finding))
+    :message (str (string/upper-case (str (first message))) (string/join (rest message)))
+    :severity (case level
+                :error 1
+                :warning 2
+                :info 3)
+    :source "clj-kondo"}))
+
+(def kondo-base-args {:cache true
+                      :cache-dir ".clj-kondo/cache"
+                      :config {:linters (into {} (map #(hash-map % {:level :off})
```

Comment:
why this level `:off`?

Reply:
A bunch of linters that overlap with clojure-lsp's existing functionalities are disabled here.